enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,329 @@
+However, they also both immediately use those VMs to start the installation process. There
+isn't an easy way to stop the current frameworks from using IPI and Assisted Installer
+after the VM creation.
+
+
+#### Launch a VM with the Ephemeral Agent Based Installer Image
+
+As a developer, I need to launch a VM with the built ephemeral agent based installer
+image so that I can test and validate this image.
+
+Both existing frameworks don't have an easy way to do this. Which makes sense because
+they predate the ephemeral agent based installer project. There also isn't an easy way
+to use an existing routine to just launch a generic ISO image which would have been
+something we could have re-purposed. 
+
+### API Extensions
+
+Not applicable at this time.
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation? What are some important details that
+didn't come across above. Go in to as much detail as necessary here. This might
+be a good place to talk about core concepts and how they relate.
+
+### Risks and Mitigations
+
+The risks for this proposal are highlighted in the motivation section. The risks we 
+have are:
+
+1) Creating a third e2e framework for installers that isn't reusable. Meaning in
+the future we will have this same issue if a new installer project comes online.
+
+2) The third e2e framework becomes very complex and is hard to understand where the 
+failures are coming from.
+
+As for how we mitigate these issues I believe that we can create a modular e2e framework
+that will mitigate this problem for future installer projects. To mitigate the second risk
+we need to make sure that each test has a clear failure message. We also need to encapsulate
+red herring messages that happen during set up and installation of components and mark these
+accordingly.
+
+## Design Details
+
+### Overall Components of the e2e Framework
+
+Using the general flow of an e2e test we can sort the overall components of the e2e framework.
+A general flow of an e2e test has the following phases:
+
+* Pre Environment Setup: 
+** Setting up VMs, networks, locations of images, image registries, etc
+* Build: 
+** Building the agent-based installer ISO and container images associated with it
+* Test:
+**  Test scenarios with pass/failure outputs
+* Reporting:
+** Reports comprised of test results
+* Environment Teardown
+** Removing VMs, networks, etc to start fresh again
+
+#### Pre Environment Setup 
+
+* The e2e framework will be a collection of Ansible playbooks and roles
+* Using roles to define reusable steps scoped to units of work
+* Use playbooks to set up common e2e workflows and chains for the project
+
+In discussions regarding this enhancement we identified the dev-scripts has components that are
+a part of the Pre Environment Setup and Environment Teardown phases of the framework. We will
+move forward with adding and potentially refactoring scripts in this repository for these phases.
```

Comment:
Did those discussions result in identifying any specific changes needed before dev-scripts could be used?

Reply:
Yes I took a stab at pointing these out.